locker

```diff
@@ -2,7 +2,7 @@
 
 function lock_image() {
 	IMAGE=$1
-	LOCKIMAGE=$(docker inspect --format='{{index .RepoDigests 0}}' $IMAGE)
+	LOCKIMAGE=`echo $1 | grep -P '.+?(?=@sha256|$)' -o`@`docker inspect --format='{{index .Id}}' $IMAGE`
```

Comment:
Lots to digest here (pun intended). A few comments:

* Why switch from `RepoDigests` to `Id`? Trying both on `openjdk:8` gives me different hashes. Can't find much documentation around the `docker inspect` output.
* `$1` should be `$IMAGE`
* Be nice to avoid grepping Perl regexes - the `man` page says they're 'highly experimental'!
* Prefer `$(...)` to backticks
* Decompose `LOCKIMAGE` into name and digest variables for readability

How about something like:

```
function lock_image() {
	IMAGE=$1
	NAME=$(echo $IMAGE | grep '^[^@]\+' -o)
	DIGEST=$(docker inspect --format='{{index .RepoDigests 0}}' $IMAGE | grep '[^@]\+$' -o)
	echo $NAME@$DIGEST
}
```

Reply:
- `RepoDigests` gave me the sha _and_ the image name, whereas `Id` gives me just the sha. I figured it would be sensible to use the one that just gave me what I needed, instead of more string wrangling!

As for splitting the function, I had that locally then changed in favour of a one line solution (closer to what I replaced!) I'll split it out again :slightly_smiling_face:

Reply:
They seem to be different things, although I don't understand the distinction:

```
$ docker inspect --format='{{index .Id}}' openjdk:8
sha256:c14ba9d23b3af968f5481a9153b7ac19920a3d22014fd4b3c6b26714203283af
$ docker inspect --format='{{index .RepoDigests 0}}' openjdk:8
openjdk@sha256:d2e5ce9a87c571481197229f618d573d183c6eca1fe3a90ab668ca1d625f0ff9
```

Ha, more is more! Thanks.

Reply:
Is it possible they're different registries that the image has been part of? I don't know enough about it.